dscript/fasta.py

```diff
@@ -1,51 +1,16 @@
-from __future__ import print_function, division
-
-
-def parse_stream(f, comment=b"#"):
-
-    name = None
-    sequence = []
-    for line in f:
-        if line.startswith(comment):
-            continue
-        line = line.strip()
-        if line.startswith(b">"):
-            if name is not None:
-                yield name, b"".join(sequence)
-            name = line[1:]
-            sequence = []
-        else:
-            sequence.append(line.upper())
-    if name is not None:
-        yield name, b"".join(sequence)
+import os
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
 
 
 def parse(f, comment="#"):
-    starter = ">"
-    empty = ""
-    if "b" in f.mode:
-        comment = b"#"
-        starter = b">"
-        empty = b""
     names = []
     sequences = []
-    name = None
-    sequence = []
-    for line in f:
-        if line.startswith(comment):
-            continue
-        line = line.strip()
-        if line.startswith(starter):
-            if name is not None:
-                names.append(name)
-                sequences.append(empty.join(sequence))
-            name = line[1:]
-            sequence = []
-        else:
-            sequence.append(line.upper())
-    if name is not None:
-        names.append(name)
-        sequences.append(empty.join(sequence))
+
+    for record in SeqIO.parse(f, "fasta"):
+        names.append(record.name)
```

Comment:
In previous versions we would take the full description line as the sequence ID. We should take just the first word (i.e. split on space). Not sure of record.name does that?

Reply:
SeqIO records parse everything before the first space as the record name, and the full line as the record description -- this should be giving the desired behavior